failure in ud_mlx5/uct_p2p_am_test.am_async_response

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11230/label=hpc-arm-hwi-jenkins,worker=3/consoleFull
```
16:34:54 [ RUN      ] ud_mlx5/uct_p2p_am_test.am_async_response/0
16:34:54 /scrap/jenkins/workspace/hpc-ucx-pr-6/label/hpc-arm-hwi-jenkins/worker/3/contrib/../test/gtest/uct/test_p2p_am.cc:447: Failure
16:34:54 Value of: m_am_count
16:34:54   Actual: 3
16:34:54 Expected: m_am_posted
16:34:54 Which is: 4
16:34:54 
[  FAILED  ] ud_mlx5/uct_p2p_am_test.am_async_response/0, where GetParam() = ud_mlx5/mlx5_0:1 (251 ms)
...
make: *** [test] Error 1
17:28:19 make: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr-6/label/hpc-arm-hwi-jenkins/worker/3/build-test/test/gtest'
```

--- a/test/gtest/uct/test_p2p_am.cc
+++ b/test/gtest/uct/test_p2p_am.cc
@@ -76,9 +76,11 @@ public:
                                 (char*)resp_req->sendbuf->ptr() + sizeof(hdr),
                                 resp_req->sendbuf->length() - sizeof(hdr));
         if (status == UCS_OK) {
+            pthread_mutex_lock(&test->m_lock);
             ++test->m_am_posted;
             resp_req->posted = true;
             delete resp_req->sendbuf;
+            pthread_mutex_unlock(&test->m_lock);
         }
         return status;
     }
@@ -306,11 +308,11 @@ protected:
         uct_p2p_am_test    *test;
         bool               posted;
     }                            m_pending_req;
+    pthread_mutex_t              m_lock;
 
 private:
     bool                         m_keep_data;
     std::vector<receive_desc_t*> m_backlog;
-    pthread_mutex_t              m_lock;
     tracer_ctx_t                 m_send_tracer;
     tracer_ctx_t                 m_recv_tracer;
 };
@@ -435,12 +437,17 @@ UCS_TEST_SKIP_COND_P(uct_p2p_am_test, am_async_response,
         ++m_am_posted;
 
         deadline = ucs_get_time() + ucs_time_from_sec(timeout);
+        pthread_mutex_lock(&m_lock);
         while ((!m_pending_req.posted || (m_am_count != m_am_posted)) &&
                (ucs_get_time() < deadline)) {
+            pthread_mutex_unlock(&m_lock);
             sender().progress();
+            pthread_mutex_lock(&m_lock);
         }
+        UCS_TEST_MESSAGE << "posted: " << m_am_posted << " am_count: " << m_am_count;
         EXPECT_TRUE(m_pending_req.posted);
         EXPECT_EQ(m_am_posted, m_am_count);
+        pthread_mutex_unlock(&m_lock);
     }
 
     status = uct_iface_set_am_handler(receiver().iface(), AM_ID, NULL, NULL, 0);

